### PR TITLE
Update docs for LLMs and Overview

### PR DESCRIPTION
**Pull request by Dev-Khant**
## Description

Update Docs:
- Add docs on how to use Mistral and Azure AI models.
- Explaining how Mem0 works.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Documentation update

## How Has This Been Tested?

- [] Unit Test
- [ ] Test Script (please provide)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

## Maintainer Checklist

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] Made sure Checks passed

### CODE REVIEW

File: docs/llms.mdx [hunk 1]
```diff
@@ -14,6 +14,8 @@ Mem0 includes built-in support for various popular large language models. Memory
   <Card title="Litellm" href="#litellm"></Card>
   <Card title="Google AI" href="#google-ai"></Card>
   <Card title="Anthropic" href="#anthropic"></Card>
+  <Card title="Mistral AI" href="#mistral-ai"></Card>
+  <Card title="Azure AI" href="#azure-ai"></Card>
```
Comment (deshraj) [inline docs/llms.mdx:18]:
```suggestion
  <Card title="Azure AI" href="#openai-azure"></Card>
```

File: docs/llms.mdx [hunk 2]
```diff
@@ -204,3 +206,57 @@ m = Memory.from_config(config)
...
+            "max_tokens": 2000,
+        }
+    }
+}
+
+m = Memory.from_config(config)
+m.add("Likes to play cricket on weekends", user_id="alice", metadata={"category": "hobbies"})
+```
+
+## Azure AI
```
Comment (deshraj) [inline docs/llms.mdx:234]:
```suggestion
## OpenAI Azure
```

**Comment by Dev-Khant** [comment]
@deshraj Comments are resolved.

File: docs/llms.mdx [hunk 3]
```diff
@@ -14,6 +14,8 @@ Mem0 includes built-in support for various popular large language models. Memory
   <Card title="Litellm" href="#litellm"></Card>
   <Card title="Google AI" href="#google-ai"></Card>
   <Card title="Anthropic" href="#anthropic"></Card>
+  <Card title="Mistral AI" href="#mistral-ai"></Card>
+  <Card title="Azure AI" href="#openai-azure"></Card>
```
Comment (deshraj) [inline docs/llms.mdx:18]:
```suggestion
  <Card title="OpenAI Azure" href="#openai-azure"></Card>
```

**Review by deshraj — APPROVED** [review]

**Pull request merged**